src/common/test-util/src/wal/kafka/topic_decorator.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cell::RefCell;
+use std::collections::HashSet;
+
+use common_config::wal::KafkaWalTopic as Topic;
+
+/// Things need to bo inserted at the front or the back of the topic.
+#[derive(Debug, Default)]
+pub enum Affix {
+    /// Inserts a provided string to each topic.
+    Fixed(String),
+    /// Computes the current time for each topic and inserts it into the topic.
+    TimeNow,
+    /// Nothing to be inserted.
+    #[default]
+    Nothing,
+}
+
+impl ToString for Affix {
+    fn to_string(&self) -> String {
+        match self {
+            Affix::Fixed(s) => s.to_string(),
+            Affix::TimeNow => chrono::Local::now().timestamp_micros().to_string(),
+            Affix::Nothing => String::default(),
+        }
+    }
+}
+
+/// Decorates a topic with the given prefix and suffix.
+pub struct TopicDecorator {
+    /// A prefix to be inserted at the front of each topic.
+    prefix: Affix,
+    /// A suffix to be inserted at the back of each topic.
+    suffix: Affix,
+    /// Topics built so far. Used to filter out duplicate topics.
+    created: RefCell<HashSet<Topic>>,
+}
+
+impl Default for TopicDecorator {
+    fn default() -> Self {
+        Self {
+            prefix: Affix::Nothing,
+            suffix: Affix::Nothing,
+            created: RefCell::new(HashSet::with_capacity(256)),
```

Comment:
Better to use `Mutex`, or change the function signature to `&mut self`

Reply:
Replaced RefCell with Mutex<Option>

Reply:
Removed the `created` field. It was designed for prevent creating duplicate topics. However, considered that a test will be run for several times, I think it's safe to remove this field.